vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -284,17 +296,54 @@ def parse_container_definition(container_def, pod_id)
 
     def parse_container(container, pod_id)
       {
-        :type          => 'ContainerKubernetes',
-        :ems_ref       => "#{pod_id}_#{container.name}_#{container.image}",
-        :name          => container.name,
-        :image         => container.image,
-        :restart_count => container.restartCount,
-        :backing_ref   => container.containerID,
-        :image_ref     => container.imageID
+        :type            => 'ContainerKubernetes',
+        :ems_ref         => "#{pod_id}_#{container.name}_#{container.image}",
+        :name            => container.name,
+        :image           => container.image,
+        :restart_count   => container.restartCount,
+        :backing_ref     => container.containerID,
+        :image_ref       => container.imageID,
+        :container_image => parse_container_image(container.image, container.imageID)
       }
+
       # TODO, state
     end
 
+    def parse_container_image(image, imageID)
+      container_image_registry, container_image = parse_image_name(image, imageID)
+      host_port = nil
+
+      unless container_image_registry.nil?
+        host_port = "#{container_image_registry[:host]}:#{container_image_registry[:port]}"
+
+        stored_container_image_registry = @data_index.fetch_path(:container_image_registry,
+                                                                 :by_host_and_port,
+                                                                 host_port)
```

Comment:
My general preference for this is:

    stored_container_image_registry = @data_index.fetch_path(
        :container_image_registry, :by_host_and_port, host_port)

But it's just personal taste (no need to change). (Same for all the other occurrences below in this method).